### PR TITLE
Net component and usage example

### PR DESCRIPTION
**Pull request by pjhenning**

### CODE REVIEW

File: content/solids/components/net.ts [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  []
+  );
+
+  return boneHinges;
+}
+/**
+  * Assumes that `a` and `b` are adjacent
+  */
+function getCommonEdge(a: Polygon, b: Polygon) {
+  return a.edges.filter(aEdge => b.edges.some(bEdge => aEdge.equals(bEdge)))[0];
```
Comment (plegner) [inline content/solids/components/net.ts:254]: I think you can do `.find()` rather than `.filter()[0]`

File: content/solids/components/net.ts [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+  if (currentFaceIndex == hinge[0]) return hinge[1];
+  else return hinge[0];
+}
+
+function mergePolys(polys: Polygon[]) {
+  const pointSet = new PointSet();
+  for (const poly of polys) {
+    pointSet.add(poly.points);
+  }
+  return new Polygon(...pointSet.points);
```
Comment (plegner) [inline content/solids/components/net.ts:288]: I don't think we need `PointSet` as a separate class. Maybe just something like this:

``` ts
function mergePolys(polys: Polygon[]) {
  const points: Point[] = [];
  for (const poly of polys) {
    for (const point of poly.points) {
      if (points.every(p => !p.equals(point))) points.push(point)
    }
  }
  return new Polygon(...point.points);
}
```

At some point, we could even make this a static `Polygon.merge()` method in euclid.js

File: content/solids/components/net.ts [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    if (isNew) this.points.push(point);
+  }
+
+  get points() {
+    return this._points;
+  }
+
+}
+
+function angleFromDegrees(val: number): Angle {
```
Comment (plegner) [inline content/solids/components/net.ts:325]: Do you want to import these two functions from euclid.js?

File: content/solids/components/net.ts [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+      skinIndices.push(boneIndex, 0, 0, 0);
+      skinWeights.push(1, 0, 0, 0);
+    }
+
+    // TODO: Change to .setAttribute if threejs version is updated
+    geometry.addAttribute('skinIndex', new THREE.Uint16BufferAttribute(skinIndices, 4));
+    geometry.addAttribute('skinWeight', new THREE.Float32BufferAttribute(skinWeights, 4));
+
+    const mat = new THREE.MeshPhongMaterial({
```
Comment (plegner) [inline content/solids/components/net.ts:94]: You could use this existing method for materials: https://github.com/mathigon/textbooks/blob/master/content/shared/components/solid.ts#L328

And add one more attribute for `skinning` that defaults to false? Just to make sure that all the materials are consistent

**Review by plegner — COMMENTED** [review]
Couple of minor comments, but otherwise this looks great!

File: content/solids/components/net.ts (hunk 3 above)
Comment (pjhenning) [reply content/solids/components/net.ts:325]: The euclid.js version (0.6.8) with these is not currently on npm as far as I can tell.

Comment (plegner) [reply content/solids/components/net.ts:296]: Oops, sorry – I forgot to click publish. Should be fixed now, but you can also change this in a later PR.

**Review by plegner — APPROVED** [review]

**Pull request merged**